src/compiler/binder.ts

```diff
@@ -1505,7 +1547,39 @@ namespace ts {
             }
         }
 
+        function bindFunctionDeclaration(node: FunctionDeclaration) {
+            if (!isDeclarationFile(file) && !isInAmbientContext(node)) {
+                if (isAsyncFunctionLike(node)) {

```

Comment:
Instead of this, what if you just looked for `await` expressions? Technically you don't need an await helper unless you use `await` right?